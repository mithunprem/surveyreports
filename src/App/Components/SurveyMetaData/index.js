import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Button, UncontrolledPopover, PopoverBody } from 'reactstrap';

const SurveyMetaData = ({ surveyMetaData }) => {

  const {
    participant_count, submitted_response_count, response_rate
  } = surveyMetaData;

  return (
    <Fragment>
      <Button className="ml-3" size="sm" color="link" 
        id="viewMetaDataButton" aria-label="View Survey Metadata Button">
        View Survey Metadata
      </Button>
      <UncontrolledPopover placement="bottom" target="viewMetaDataButton">
        <PopoverBody>
          <p>
            Participant Count : {participant_count}<br />
            Submitted Response Count : {submitted_response_count}<br />
            Response Rate : {(response_rate*100).toFixed(2)}%
          </p>
        </PopoverBody>
      </UncontrolledPopover>
    </Fragment>
  )
}

SurveyMetaData.propTypes = {
  surveyMetaData: PropTypes.object
}

export default SurveyMetaData;
