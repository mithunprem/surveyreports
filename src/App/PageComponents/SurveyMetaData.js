import React, { Component, Fragment } from 'react';
import {
  Col, Card, CardHeader, CardBody, CardText, Button
} from 'reactstrap';

export default class SurveyMetaData extends Component {

  render() {
    const {
      name, participant_count, url,
      submitted_response_count, response_rate
    } = this.props.survey;

    return (
      <Fragment>
          <Col xs={12} sm="auto" className="p-3">
            <Card>
              <CardHeader tag="h5">{name}</CardHeader>
              <CardBody>
                <CardText>Participant Count : {participant_count}</CardText>
                <CardText>Submitted Response count : {submitted_response_count}</CardText>
                <CardText>Response Rate : {response_rate.toFixed(2)}%</CardText>
                <Button>View Survey Details</Button>
              </CardBody>
            </Card>
          </Col>
      </Fragment>
    )
  }
}
