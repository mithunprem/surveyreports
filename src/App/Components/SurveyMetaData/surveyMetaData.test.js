import React from 'react';
import { create } from 'react-test-renderer';
import SurveyMetaData from '../../Components/SurveyMetaData';

describe("SurveyMetaData", () => {

  const surveyMetaDataSample = {
    name: "Simple Survey",
    participant_count: 6,
    response_rate: 0.8333333333333334
  };

  test("should render without throwing an error", () => {
    const div = document.createElement('div');
    div.setAttribute("id", "viewMetaDataButton");
    document.body.appendChild(div);

    const surveyMetaData = create(<SurveyMetaData surveyMetaData={surveyMetaDataSample}  />).toJSON();
    expect(surveyMetaData).toMatchSnapshot();
  });
});
