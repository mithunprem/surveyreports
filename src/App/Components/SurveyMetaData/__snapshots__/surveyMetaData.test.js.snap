// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveyMetaData should render without throwing an error 1`] = `
<button
  aria-label={null}
  className="ml-3 btn btn-link btn-sm"
  id="viewMetaDataButton"
  onClick={[Function]}
>
  View Survey Metadata
</button>
`;
